src/main/java/org/commcare/suite/model/MenuLoader.java

```diff
@@ -73,7 +73,7 @@ private void getMenuDisplayables(CommCarePlatform platform,
             for (Menu m : s.getMenus()) {
                 try {
                     if (m.getId().equals(menuID)) {
-                        if (menuIsRelevant(sessionWrapper, m)) {
+                        if (menuIsRelevant(sessionWrapper, m) && menuAssertionsPass(sessionWrapper, m)) {
```

Comment:
Having this here seems like it will behave exactly like menu relevancy conditions. I thought this would only trigger when you select a menu and that the assertion text would get shown to the user.

Reply:
after adding in the error handling here 7797be3e93df0517bdbb0e5a8936c4128ade02ce this seems to work as expected. I tested locally with @esoergel's with custom assertions set up.

Reply:
> Having this here seems like it will behave exactly like menu relevancy conditions. I thought this would only trigger when you select a menu and that the assertion text would get shown to the user.

@snopoke This method only gets invoked on clicking a menu. The `menuIsRelevant(sessionWrapper, m)` is more of a safety check here and in reality user would not be able to see this menu in the first place if it's not relevant.